components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.m

```diff
@@ -0,0 +1,657 @@
+  CGFloat topAreaInsetForHeader = (self.headerViewController ? MDCDeviceTopSafeAreaInset() : 0);
+  CGFloat drawerOffset = self.contentHeaderTopInset - topAreaInsetForHeader;
+  CGFloat headerHeightWithoutInset = self.contentHeaderHeight - topAreaInsetForHeader;
+  CGFloat contentDiff = contentYOffset - drawerOffset;
+  CGFloat maxScrollOrigin = self.trackingScrollView.contentSize.height -
+                            self.presentingViewBounds.size.height + headerHeightWithoutInset;
+  BOOL scrollingUpInFull = contentDiff < 0 && self.trackingScrollView.bounds.origin.y > 0;
+  if (self.scrollView.bounds.origin.y >= drawerOffset || scrollingUpInFull) {
+    // Update the main content view's scrollView offset
+    CGRect mainContentViewBounds = self.trackingScrollView.bounds;
+    mainContentViewBounds.origin.y += contentDiff;
+    mainContentViewBounds.origin.y = MIN(maxScrollOrigin, MAX(mainContentViewBounds.origin.y, 0));
+    self.trackingScrollView.bounds = mainContentViewBounds;
+
+    // If we reach full screen or if we are scrolling up after being in full screen.
+    if (self.trackingScrollView.bounds.origin.y < maxScrollOrigin || scrollingUpInFull) {
+      // If we still didn't reach the end of the content, or if we are scrolling up after reaching
+      // the end of the content.
+
+      // Update the drawer's scrollView's offset to be static so the content will scroll instead.
+      CGRect scrollViewBounds = self.scrollView.bounds;
+      scrollViewBounds.origin.y = drawerOffset;
+      self.scrollView.bounds = scrollViewBounds;
+
+      // Make sure the drawer's scrollView's content size is the full size of the content
+      CGSize scrollViewContentSize = self.presentingViewBounds.size;
+      scrollViewContentSize.height += self.contentHeightSurplus;
+      self.scrollView.contentSize = scrollViewContentSize;
+    } else {
+      // Have the drawer's scrollView's content size be static so it will bounce when reaching the
+      // end of the content.
+      CGSize scrollViewContentSize = self.scrollView.contentSize;
+      scrollViewContentSize.height =
+          drawerOffset + self.scrollView.frame.size.height + 2 * topAreaInsetForHeader;
+      self.scrollView.contentSize = scrollViewContentSize;
+    }
+  }
+}
+
+- (void)addScrollViewObserver {
+  if (self.scrollViewObserved) {
+    return;
+  }
+  self.scrollViewObserved = YES;
+  [self.scrollView addObserver:self
+                    forKeyPath:kMDCBottomDrawerContentOffsetKeyPath
+                       options:NSKeyValueObservingOptionNew | NSKeyValueObservingOptionOld
+                       context:nil];
+}
+
+- (void)removeScrollViewObserver {
+  if (!self.scrollViewObserved) {
+    return;
+  }
+  self.scrollViewObserved = NO;
+  [self.scrollView removeObserver:self forKeyPath:kMDCBottomDrawerContentOffsetKeyPath];
+}
+
+#pragma mark UIViewController
+
+- (void)viewDidLoad {
+  [super viewDidLoad];
+  [self setUpContentHeader];
```

Comment:
Please add space after `[super viewDidLoad]` and all calls to `super`.

Reply:
done